### ISSUE TITLE
Adding a Collection Association on a Class Diagram shows an error

### ISSUE THREAD

**Issue by davkean** [dotnet/project-system#3003]
1. Create a new Console App (.NET Core)
2. Right-click project -> Add -> New Item -> Class Diagram
3. Add a new source file with the following contents:

``` C#
    internal interface IVsHierarchy
    {
        string[] Foo { get; }
    }
```

4. From solution explorer, drag the file onto the class diagram
5. Right-click on Foo property in the class diagram and choose "Show as Collection Association"

Expected: For the property to show an collection association
Actual:

```
---------------------------
Microsoft Visual Studio
---------------------------
This is not a valid type name.
---------------------------
OK   Help   
---------------------------
```

**Comment by davkean** [comment]
Looks like under the covers this is occurring because the diagram thinks we're a VB class diagram, probably related to: https://github.com/dotnet/project-system/issues/2999

**Comment by wjfamilia** [comment]
Following your other [#256 thread](https://github.com/dotnet/project-system/issues/256) I was able to get the Class Designer working in my .NET Core 2.0 solution by modifying the Microsoft.CSharp.DesignTime.targets file as suggested, but the "Show as Collection Association" is a vital feature to create a clean class diagram. I hope a fix can be found and I appreciate the effort so far.

**Comment by yesitspeter** [comment]
I am having this same issue.

**Comment by jagmagh** [comment]
Having the exact same issue as well. Unable to visualize and collection associations - which is one of the most common associations between classes  - does make this a bit of a show stopper in terms of using the Class diagramming tool. Any workarounds, anyone?

**Comment by Pilchie** [comment]
See also :link: https://developercommunity.visualstudio.com/content/problem/229341/adding-generic-interfaces-to-class-diagrams-in-net.html, which is another scenario with the same error message.

**Comment by DJDaemonix** [comment]
I would like to request a status update on this one. Has a workaround been found / fix rolled out ?
I came across this problem alongside #2999, thinking they might be related.

**Comment by davkean** [comment]
Microsofties, I've an internal PR for this here: https://devdiv.visualstudio.com/DevDiv/_git/CPS/pullrequest/122409?_a=files.

**Comment by davkean** [comment]
I've confirmed with the above PR applied, this now works:

![image](https://user-images.githubusercontent.com/1103906/40160873-58004e78-59f2-11e8-80c0-0e923c4426ff.png)

This bug has the same underlying cause as https://github.com/dotnet/project-system/issues/2999 and hence going to dup it against it.

**Comment by davkean** [comment]
This exactly issue will be fixed in 15.8 Preview 3, however, make note there's still one issue that we're aware of which is called out here: https://github.com/dotnet/project-system/issues/2999#issuecomment-389762981.
